src/core/meshcop/border_agent.cpp

```diff
@@ -447,6 +447,7 @@ template <>
 void BorderAgent::HandleTmf<kUriCommissionerPetition>(Coap::Message &aMessage, const Ip6::MessageInfo &aMessageInfo)
 {
     IgnoreError(ForwardToLeader(aMessage, aMessageInfo, kUriLeaderPetition));
+    mTimer.Start(kKeepAliveTimeout);
```

Comment:
This now starts the timer when the BA forwards the "petition".

Should we also remove the starting of timer in `BorderAgent::HandleConnected()`?

Reply:
Rescoped this PR for log enhancements only for the discussion https://github.com/openthread/openthread/pull/10549#issuecomment-2264930711